sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -94,47 +92,14 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
     Configuration configuration = taskAttemptContext.getConfiguration();
     ParquetInputSplit split = (ParquetInputSplit)inputSplit;
     this.file = split.getPath();
-    long[] rowGroupOffsets = split.getRowGroupOffsets();
-
-    ParquetMetadata footer;
-    List<BlockMetaData> blocks;
-
-    // if task.side.metadata is set, rowGroupOffsets is null
-    if (rowGroupOffsets == null) {
```

Comment:
Yeah, let's add an assert. That sounds like a good idea.

Reply:
OK~ Will  do it later. a little busy at the moment

Reply:
2a57f30 add a assert use `Preconditions.checkState`